cscs-checks/apps/tensorflow/tensorflow_check.py

```diff
@@ -0,0 +1,72 @@
+import os
+import itertools
+import reframe.utility.sanity as sn
+
+from reframe.core.pipeline import RunOnlyRegressionTest
+
+
+class TensorFlowTestBase(RunOnlyRegressionTest):
+    def __init__(self, model_name, **kwargs):
+        super().__init__('tensorflow_%s_check' % model_name,
+                         os.path.dirname(__file__), **kwargs)
+
+        self.descr = 'Tensorflow official %s test' % model_name
+        self.strict_check = False
+        self.valid_systems = ['daint:gpu', 'dom:gpu']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.sourcesdir = 'https://github.com/tensorflow/models.git'
+        self.maintainers = ['TM']
+        self.tags = {'production'}
+        self.num_tasks = 1
+        self.num_gpus_per_node = 1
+        self.modules = ['TensorFlow/1.4.1-CrayGNU-17.08-cuda-8.0-python3']
```

Comment:
Why not use as module name `TF/1.4.1-python3` and map the module name to the real module?
Probably this way, we could extend it to get `1.4.1` as a parameter and do the proper `git checkout` and have different TF modules versions mapped.

What do you think?

Reply:
I guess that in order to achieve this a corresponding file with the actual mappings has to be created. This needs also changes in the ci to use the corresponding command line argument. @vkarak what do you think?

Reply:
Decoupling the module names in the tests from the real modules, I see it as a separate project. I don't think we should address it specifically for this test.